src/Models/UserPermission.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Salt\Core\Models;
+
+use Illuminate\Database\Eloquent\Relations\Pivot;
+use Salt\Core\Observers\PermissionDependenciesObserver;
+
+class UserPermission extends Pivot
+{
+    protected $table = 'permission_user';
```

Comment:
UserPermission needed to be broken out into its own class in order to add an observer